historical/attributes.py

```diff
@@ -50,13 +58,39 @@ def serialize(self, value):
             return value
         return value.strftime(DATETIME_FORMAT)
 
+
+def decimal_default(obj):
+    if isinstance(obj, decimal.Decimal):
+        return float(obj)
+    raise TypeError
+
+
+class HistoricalDecimalAttribute(Attribute):
+    """
+    A number attribute
+    """
+    attr_type = NUMBER
+
+    def serialize(self, value):
+        """
+        Encode numbers as JSON
+        """
+        return json.dumps(value, default=decimal_default)
+
     def deserialize(self, value):
         """
-        Takes a iso8601 datetime string and returns a datetime object
+        Decode numbers from JSON
         """
-        return datetime.strptime(value, DATETIME_FORMAT)
+        return json.loads(value)
 
 
-class TagAttribute(MapAttribute):
-    Key = UnicodeAttribute()
-    Value = HistoricalUnicodeAttribute()
+pynamodb.attributes.SERIALIZE_CLASS_MAP = {
```

Comment:
@mikegrima I think I may have solved a few issues. I patch the pynamodb attribute map to map to our own classes. Essentially what this means is that we can ensure our attribute is called when nothing is specified (deeply nested maps). This works for both the empty string case, and the decimal case. 

In the empty string case I chose to simple replace `''` with `'<empty>` and ensure that it gets correctly deserialized back to `''`. This might take some communication to external users, should be transparent to us.